### DISCUSSION TITLE
`gix-config` and `gix-config-value` how to read or write a given `[section] key=value`?

### DISCUSSION THREAD

**Discussion by moto-timo** [GitoxideLabs/gitoxide#1324] [Q&A]
I was looking for how to read (or write) a value for a given `[section] key=value`, for instance`[user] name`.

I realized at least for `User::NAME` we already have [several examples](https://github.com/search?q=repo%3AByron%2Fgitoxide+%22User%3A%3ANAME%22&type=code):
        `let (user_name, mut user_email) = entity_in_section(config, &User::NAME, &User::EMAIL, None);`

I am curious what your thoughts might be about implementing `gix config --get user.name`  `gix config --set user.name` or something like that.

Also, we probably want a `gix config --path .gitmodules` or similar argument/option to allow plumbing interaction with arbitrary git config compliant file types.

**Comment by Byron** [comment]
`gix-config` has methods to [read](https://docs.rs/gix-config/latest/gix_config/struct.File.html#method.string) and [write](https://docs.rs/gix-config/latest/gix_config/struct.File.html#method.set_raw_value) values in various ways, and `gix` has [abstractions](https://docs.rs/gix/latest/gix/config/struct.SnapshotMut.html#method.set_value) for that as well. 

As `gix` is a developer tool, and to my mind there is no need to try to make it into something that rivals `git` in features for example, or try to provide an alternative to certain `git` commands necessarily.

Particularly with the example above, I don't see what the value is as `git` can already do that and `gix` can do it too - there are many, many tests that validate it. Thus, building more `gix config` capabilities seems more like an exercise than solving an actual problem.

What am I missing?
